### ISSUE TITLE
Error on shutdown unmounting /var/log

### ISSUE THREAD

**Issue by r-vdp** [WilliButz/preservation#2]
I get the following error on shutdown
```
sep 07 15:11:57 framework systemd[1]: var-log.mount: Mount process exited, code=exited, status=32/n/a
sep 07 15:11:57 framework systemd[1]: Failed unmounting var-log.mount.
```
Since it's at shutdown, I can't inspect the system state when this happens. Probably we'd need some ordering to make sure that we only unmount this after journalctl stopped?

I configured it like in the examples, with `{ directory = "/var/log"; inInitrd = true; }`

**Comment by WilliButz** [comment]
Can confirm this.
The cause for the attempted unmount should be `Conflicts=umount.target` in the mount units, which is started on shutdown ([bootup(7)](https://www.freedesktop.org/software/systemd/man/latest/bootup.html#System%20Manager%20Shutdown)). I haven't gotten around to figuring out how to handle this case with systemd-journald.service correctly

**Comment by WilliButz** [comment]
Something like this should precede the above log messages:
```
systemd[1]: Unmounting var-log.mount...
umount[112539]: umount: /var/log: target is busy.
```

**Comment by WilliButz** [comment]
`var-log.mount` is ordered `Before=systemd-journal-flush.service`:
```
→ systemctl show -p Before var-log.mount
Before=systemd-update-utmp.service umount.target systemd-journal-flush.service local-fs.target
```
meaning it should be stopped after the flush service on shutdown ([systemd.unit(5)](https://www.freedesktop.org/software/systemd/man/latest/systemd.unit.html#Before=)):
> When two units with an ordering dependency between them are shut down, the inverse of the start-up order is applied.

So this should be executed before `/var/log` is unmounted:

```
# /etc/systemd/system/systemd-journal-flush.service
ExecStop=journalctl --smart-relinquish-var
```

From [journalctl(1)](https://www.freedesktop.org/software/systemd/man/latest/journalctl.html#--smart-relinquish-var) about `--smart-relinquish-var`:
> This operation is used during system shutdown in order to make the journal daemon stop writing data to /var/log/journal/ in case that directory is located on a mount point that needs to be unmounted.


However, I can't find anything in my logs about the flush service being stopped, which doesn't seem right

**Comment by WilliButz** [comment]
Fixed by 97104c042dfa36e24659db05976409294d774ee5 (see https://github.com/WilliButz/preservation/pull/3#issuecomment-2453109847). Closing
